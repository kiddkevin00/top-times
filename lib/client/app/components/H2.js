import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import '../../static/assets/styles/components/h2.scss';

// eslint-disable-next-line react/prefer-stateless-function
export default class H2 extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
  };

  static defaultProps = {
    className: '',
  };

  render() {
    const className = cx('pn-h2', this.props.className);

    return <h2 className={className}>{this.props.children}</h2>;
  }
}
