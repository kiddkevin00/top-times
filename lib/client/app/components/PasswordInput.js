import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import '../../static/assets/styles/components/passwordInput.scss';

class PasswordInput extends Component {
  static propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    placeholder: PropTypes.string,
  };

  static defaultProps = {
    placeholder: 'Password',
  };

  state = { showPassword: false };

  toggleVisibility = event => {
    event.stopPropagation();

    this.setState({
      showPassword: !this.state.showPassword,
    });
  };

  render() {
    const type = this.state.showPassword ? 'text' : 'password';
    const { value, onChange, placeholder, ...otherProps } = this.props;
    const passwordVisibilityToggleClassName = cx('password-visibility-toggle', {
      enabled: value && value.length,
      disabled: !(value && value.length),
    });

    return (
      <div className="password-input-container">
        <input
          type={type}
          value={value}
          onChange={onChange}
          placeholder={placeholder}
          {...otherProps}
        />
        {/* eslint-disable jsx-a11y/interactive-supports-focus, jsx-a11y/click-events-have-key-events */}
        <span
          onClick={this.toggleVisibility}
          className={passwordVisibilityToggleClassName}
          role="button"
        >
          {!this.state.showPassword ? 'Show' : 'Hide'}
        </span>
        {/* eslint-enable jsx-a11y/interactive-supports-focus, jsx-a11y/click-events-have-key-events */}
      </div>
    );
  }
}

export { PasswordInput as default };
