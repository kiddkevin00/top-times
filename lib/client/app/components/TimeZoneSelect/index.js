import SelectTimeZone from './SelectTimeZone';
import { generateUniqueId } from '../../utils/helpers';
import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import '../../../static/assets/styles/components/timeZoneSelect.scss';

export default function TimeZoneSelect({ className, id, onChange, value }) {
  const clazzName = cx('time-zone-select', className);

  return (
    <SelectTimeZone
      className={clazzName}
      id={id || generateUniqueId()}
      onChange={onChange}
      value={value}
    />
  );
}

TimeZoneSelect.propTypes = {
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string,
  id: PropTypes.string,
  value: PropTypes.string,
};

TimeZoneSelect.defaultProps = {
  className: '',
  id: undefined,
  value: undefined,
};

export const { nullValueInTimeZoneSelect } = SelectTimeZone;
