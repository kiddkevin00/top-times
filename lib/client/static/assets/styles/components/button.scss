@import '../_modules/variables';

.pn-button {
  max-width: 100%;
  width: 366px;
  height: 48px;
  border-radius: 24px;
  font-size: 16px;
  font-weight: bold;
  color: white;
  background-color: $tint-color;

  &:disabled {
    opacity: 0.5;
    cursor: default;
  }
}
