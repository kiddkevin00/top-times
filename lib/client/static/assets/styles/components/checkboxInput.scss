.pn-checkbox {
  display: flex;
  align-items: center;
  margin: 20px 0;
  cursor: pointer;

  .checked-mark-container {
    flex-basis: 20px;
    flex-shrink: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgba(204, 208, 213, 0.3);
    border: 1px solid rgb(204, 208, 213);
    border-radius: 4px;
    height: 20px;
  }

  p {
    text-align: left;
    margin: 0 0 0 16px;
    font-size: 15px;
  }
}
