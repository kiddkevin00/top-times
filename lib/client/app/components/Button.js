import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import '../../static/assets/styles/components/button.scss';

// eslint-disable-next-line react/prefer-stateless-function
export default class Button extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    type: PropTypes.string,
    disabled: PropTypes.bool,
  };

  static defaultProps = {
    className: '',
    type: 'button',
    disabled: false,
  };

  render() {
    const className = cx('pn-button', this.props.className);

    return (
      <button
        className={className}
        onClick={this.props.onClick}
        type={this.props.type}
        disabled={this.props.disabled}
      >
        {this.props.children}
      </button>
    );
  }
}
