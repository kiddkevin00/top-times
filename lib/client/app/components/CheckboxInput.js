import { FaCheck as CheckedMark } from 'react-icons/fa/';
import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import '../../static/assets/styles/components/checkboxInput.scss';

export default class CheckboxInput extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onChange: PropTypes.func.isRequired,
    value: PropTypes.bool.isRequired,
    className: PropTypes.string,
  };

  static defaultProps = {
    className: '',
  };

  onKeyPress = event => {
    if (event.key === 'Enter') {
      this.props.onChange(!this.props.value);
    }
  };

  render() {
    const { className, onChange, value, children } = this.props;
    const clazzName = cx('pn-checkbox', className);

    return (
      <div
        className={clazzName}
        onClick={() => onChange(!value)}
        role="button"
        tabIndex="-1"
        onKeyPress={this.onKeyPress}
      >
        <div className="checked-mark-container">
          {value ? <CheckedMark /> : null}
        </div>

        <p>{children}</p>
      </div>
    );
  }
}
