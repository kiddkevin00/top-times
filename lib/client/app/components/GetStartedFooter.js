import H2 from './H2';
import Button from './Button';
import { withRouter } from 'react-router-dom';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import '../../static/assets/styles/components/getStartedFooter.scss';

class GetStartedFooter extends Component {
  static propTypes = {
    history: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  };

  goToLogin = () => {
    this.props.history.push('/login');
  };

  render() {
    return (
      <section className="get-started-footer">
        <div>
          <H2 className="get-started-h2">
            You don't have to wrestle with timezones every single time anymore
          </H2>
          <Button className="get-started-btn" onClick={this.goToLogin}>
            LETS GET STARTED
          </Button>
        </div>
      </section>
    );
  }
}

export { GetStartedFooter as UnwrappedGetStartedFooter };
export default withRouter(GetStartedFooter);
