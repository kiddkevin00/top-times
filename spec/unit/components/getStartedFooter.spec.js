import { UnwrappedGetStartedFooter } from '../../../lib/client/app/components/GetStartedFooter';
import Button from '../../../lib/client/app/components/Button';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Get Started Footer component', () => {
  let props;

  beforeEach(() => {
    props = {
      history: { push: jest.fn() },
    };
  });

  it('should be rendered correctly', () => {
    const wrapper = shallow(<UnwrappedGetStartedFooter {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should always have predefined CSS classes', () => {
    const wrapper = shallow(<UnwrappedGetStartedFooter {...props}>click me</UnwrappedGetStartedFooter>);

    expect(wrapper.find('.get-started-footer').length).toBe(1);
  });

  it('navigate to Login page when clicking on action button', () => {
    const wrapper = shallow(<UnwrappedGetStartedFooter {...props}>click me</UnwrappedGetStartedFooter>);

    wrapper.find(Button).simulate('click');

    expect(props.history.push).toHaveBeenCalledTimes(1);
    expect(props.history.push).toHaveBeenCalledWith('/login');
  });
});
