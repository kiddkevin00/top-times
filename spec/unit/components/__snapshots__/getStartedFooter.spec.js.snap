// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Started Footer component should be rendered correctly 1`] = `
<section
  className="get-started-footer"
>
  <div>
    <H2
      className="get-started-h2"
    >
      You don't have to wrestle with timezones every single time anymore
    </H2>
    <Button
      className="get-started-btn"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      LETS GET STARTED
    </Button>
  </div>
</section>
`;
